examples/spring-boot/build.gradle

```diff
@@ -0,0 +1,23 @@
+buildscript {
+    ext {
+        springBootVersion = '2.1.2.RELEASE'
+    }
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
+    }
+}
+
+apply plugin: 'org.springframework.boot'
+apply plugin: 'io.spring.dependency-management'
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
+    implementation 'org.springframework.boot:spring-boot-starter-data-redis'
+    implementation 'org.springframework.boot:spring-boot-starter-web'
+    runtimeOnly 'org.postgresql:postgresql'
+    testImplementation 'org.springframework.boot:spring-boot-starter-test'
+    testCompile project(':postgresql')
```

Comment:
Users won't be able to execute the sample as-is. This project relies on other modules of the build. Optimally, you'd inject the TestContainer code "under test" at runtime with TestKit. I'd have to try this out myself but I think it's possible.

Reply:
I think that's a good first step. I see the following benefits:

- Developers of this project can use auto-completion as each example project is part of the multi-project builds.
- Users of TestContainers can copy/paste the example code and it should work as-is.

There's one thing I'd personally be missing:

- You can really deeply test example code from a user's perspective. I think it would be good if you could run an example project as a user would and then verify a specific outcome (e.g. tests finish successfully or the test console output should contain x).

Overall, I am happy with your approach. It will unblock other documentation effort. BTW: Do your changes also use the same approach for the quickstart examples?

Reply:
> * I think it would be good if you could run an example project as a user would and then verify a specific outcome (e.g. tests finish successfully or the test console output should contain x).

We do run them as a black box (think `./gradlew test`). More specific tests (e.g. for the standard output, or env variables / properties) **IMO** should be done as part of the integration testing suite of the core module

> BTW: Do your changes also use the same approach for the quickstart examples?

We plan to merge both under the same "examples" set of projects